src/loaders/types.js

```diff
@@ -8,20 +8,22 @@ import CellSetsJsonLoader from './CellSetsJsonLoader';
 import AnnDataLoaders from './anndata-loaders';
 import GenomicProfilesZarrLoader from './GenomicProfilesZarrLoader';
 
+import { AnnDataSource, ZarrDataSource } from './data-sources';
+
 const ANNDATA = 'anndata';
 
 export const fileTypeToLoader = {
-  'expression-matrix.zarr': MatrixZarrLoader,
+  'expression-matrix.zarr': [ZarrDataSource, MatrixZarrLoader],
   'clusters.json': ClustersJsonAsMatrixZarrLoader,
   'genes.json': GenesJsonAsMatrixZarrLoader,
   'cells.json': JsonLoader,
   'molecules.json': JsonLoader,
   'neighborhoods.json': JsonLoader,
   'raster.json': RasterJsonLoader,
```

Comment:
I think that makes sense, but let's shoot for trying that out in a separate PR!

The thing I'd like to get most feedback on is **what fields should a DataSource constructor should have access to?**

For example, the `JsonLoader` has access to `url`, `requestInit`, `type` and `fileType`. Should the loader handle validation or the data source? In that case, the loader would need to know the `fileType`; but for zarr, we want to avoid sharing the `fileType` because the data source captures multiple `fileType`s.

Reply:
@manzt This would exactly be it.  The source only cares about the `url` and `requestInit`, and there only because it is needed for fetching (from) the `url`.